### PR TITLE
Spec changes

### PR DESCRIPTION
**Pull request by CesarGallego**
Minor changes in order to simplify.

### CODE REVIEW

File: spec/README.md [hunk 1]
```diff
@@ -126,15 +127,18 @@ whole lifetime of the server.
 
 ## Design Principles
 
+* TCP implementations should follow a general principle of robustness: be
```
Comment (nilp0inter) [inline spec/README.md]: Kapow! instead of TCP

File: spec/README.md [hunk 2]
```diff
@@ -126,15 +127,18 @@ whole lifetime of the server.
 
 ## Design Principles
 
+* TCP implementations should follow a general principle of robustness: be 
+  conservative in what you do, be liberal in what you accept from others.
 * We reuse conventions of well-established software projects, such as Docker.
 * All requests and responses will leverage JSON as the data encoding method.
-* The API calls responses will have two distinct parts:
+* The API calls responses will several parts:
```
Comment (nilp0inter) [inline spec/README.md]: responses have several parts

File: spec/README.md [hunk 3]
```diff
@@ -126,15 +127,18 @@ whole lifetime of the server.
...
   * The HTTP status code (e.g., `400`, which is a bad request).  The target
     audience of this information is the client code.  The client can thus use
     this information to control the program flow.
   * The HTTP reason phrase.  The target audience in this case is the human
     operating the client.  The human can use this information to make a
     decision on how to proceed.
+  * The body it's optional
```
Comment (nilp0inter) [inline spec/README.md]: is optional

File: spec/README.md [hunk 4]
```diff
@@ -176,7 +180,8 @@ respond to an external event (e.g.  an incoming HTTP request).
 
 #### List routes
 
-Returns JSON data about the current routes.
+Returns JSON with all data about the current routes. Be aware that the command
```
Comment (nilp0inter) [inline spec/README.md]:
```suggestion
Returns JSON with all the data about the current routes. Be aware that the command
```

File: spec/README.md [hunk 5]
```diff
@@ -239,19 +246,7 @@ Accepts JSON data that defines a new route to be appended to the current routes.
     ```
 * **Error Responses**:
   * **Code**: `400 Malformed JSON`
-  * **Code**: `422 Invalid Data Type`
   * **Code**: `422 Invalid Route Spec`
```
Comment (nilp0inter) [inline spec/README.md]:
```suggestion
  * **Code**: `422 Invalid Route`
```

File: spec/README.md [hunk 6]
```diff
@@ -303,20 +299,7 @@ Accepts JSON data that defines a new route to be appended to the current routes.
...
```
Comment (nilp0inter) [inline spec/README.md]:
```suggestion
  * **Code**: `422 Invalid Route`
```

File: spec/README.md (hunk 1 above)
Comment (CesarGallego) [reply spec/README.md]: Doing

File: spec/README.md (hunk 2 above)
Comment (CesarGallego) [reply spec/README.md]: Doing

File: spec/README.md (hunk 3 above)
Comment (CesarGallego) [reply spec/README.md]: Doing

**Pull request merged**